ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,292 @@
+  $header = array(
+    array(
+      'data' => t('Order ID'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Transaction ID'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Date'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('User'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Text'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Amount'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Cleared'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Paid'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Captured'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => '',
+      'class' => 'payment',
+    ),
+  );
+  foreach ($report_valid as $record) {
+    $status = $record->payment_status * 1 . $record->provider_payment_status * 1 . $record->dibs_capture_status * 1;
+    switch ($status) {
+      case '111':
+      case '000':
+        $status = 'ok';
+        break;
+
+      case '100':
+        $status = 'warning';
+        break;
+
+      default:
+        $status = 'error';
+    }
+    $order_info_long = unserialize($record->order_info_long);
+    if (is_array($order_info_long) && count($order_info_long) > 0) {
+      $text = implode(array_shift($order_info_long)) . ' ';
+      $text .= implode(', ', array_map('array_shift', $order_info_long));
+    }
+    elseif (is_string($order_info_long)) {
+      $text = $order_info_long;
+    }
+    else {
+      $text = $record->order_info_short;
+    }
+    $row = array(
+      array(
+        'data' => check_plain($record->payment_order_id),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->payment_transaction_id),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->payment_time),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->customer_uid),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($text),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->payment_price),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => $record->payment_status ? t('yes') : t('no'),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => $record->provider_payment_status ? t('yes') : t('no'),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => $record->dibs_capture_status ? t('yes') : t('no'),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => '',
+        'class' => 'icon ' . $status,
+      ),
+    );
+    $rows[] = $row;
+    $no_transactions_msg = FALSE;
+  }
+
+  $output = theme('table', array(
+    'header' => $header,
+    'rows' => $rows,
+  ));
+  if ($no_transactions_msg) {
```

Comment:
This seems unneeded. If there are no transactions the function returns on line L102.